app/dialog/cocoa/dlg.m

```diff
@@ -109,12 +114,29 @@ - (NSInteger)runPanel:(NSSavePanel*)panel {
 	if(self->params->title != nil) {
 		[panel setTitle:[[NSString alloc] initWithUTF8String:self->params->title]];
 	}
+	// Use modern allowedContentTypes API on macOS 11+ for better file support
+	if(self->params->numext > 0) {
+		if(@available(macOS 11.0, *)) {
```

Comment:
Ollama supports macOS 13+ and so I believe we only need to keep this branch of the code